### PR TITLE
statistics: the overflow check is not correct in `0-Math.MinInt64` situation.

### LINKED ISSUE #11610: An overflow int64 error in range calculation.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
```sql
tidb(localhost:4000) > create table t(a int, b int, primary key(a));
Query OK, 0 rows affected (0.01 sec)

tidb(localhost:4000) > insert into t values(1,1),(2,2),(3,3);
Query OK, 3 rows affected (0.00 sec)
Records: 3  Duplicates: 0  Warnings: 0

tidb(localhost:4000) > analyze table t;
Query OK, 0 rows affected (0.01 sec)

tidb(localhost:4000) > desc select 1 from t where (t.a is not null) and (t.a <= 0);
```

2. What did you expect to see?
Query successfully, return an empty set.


3. What did you see instead?
Panic, the error happened in `statistics.enumRangeValues`.


4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```sql
tidb(localhost:4000) > select tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                                         |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-alpha-20-g03bb8d781-dirty
Git Commit Hash: 03bb8d7818b85996d6a85664223857a4b395030e
Git Branch: HEAD
UTC Build Time: 2019-08-05 06:39:50
GoVersion: go version go1.12 darwin/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
fix #11610.

### What is changed and how it works?
Change the check range.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - None

Side effects

 - None

Related changes

 - None

### CODE REVIEW

File: statistics/scalar.go [hunk 1]
```diff
@@ -195,7 +195,7 @@ func enumRangeValues(low, high types.Datum, lowExclude, highExclude bool) []type
 	case types.KindInt64:
 		// Overflow check.
 		lowVal, highVal := low.GetInt64(), high.GetInt64()
-		if lowVal < 0 && highVal > 0 {
+		if lowVal <= 0 && highVal >= 0 {
```
Comment (lamxTyler) [inline statistics/scalar.go]: It may miss some valid ranges, for example [-10, 0).

**Comment by XuHuaiyu** [comment]
Does this PR need the cherry-pick label?

**Comment by lzmhhh123** [comment]
@XuHuaiyu No need cherry-pick, it's a new bug in master and not in any release branch.

**Pull request merged**